fdbclient/vexillographer/fdb.options

```diff
@@ -148,16 +148,20 @@ description is not currently required but encouraged.
-            description="Set a timeout in milliseconds which, when elapsed, will cause each transaction automatically to be cancelled. This sets the ``timeout`` option of each transaction created by this database. See the transaction option description for more information. Using this option requires that the API version is 610 or higher." />
+            description="Set a timeout in milliseconds which, when elapsed, will cause each transaction automatically to be cancelled. This sets the ``timeout`` option of each transaction created by this database. See the transaction option description for more information. Using this option requires that the API version is 610 or higher." 
+            defaultFor="500"/>
     <Option name="transaction_retry_limit" code="501"
             paramType="Int" paramDescription="number of times to retry"
-            description="Set a timeout in milliseconds which, when elapsed, will cause a transaction automatically to be cancelled. This sets the ``retry_limit`` option of each transaction created by this database. See the transaction option description for more information." />
+            description="Set a timeout in milliseconds which, when elapsed, will cause a transaction automatically to be cancelled. This sets the ``retry_limit`` option of each transaction created by this database. See the transaction option description for more information." 
+            defaultFor="501"/>
     <Option name="transaction_max_retry_delay" code="502"
             paramType="Int" paramDescription="value in milliseconds of maximum delay"
-            description="Set the maximum amount of backoff delay incurred in the call to ``onError`` if the error is retryable. This sets the ``max_retry_delay`` option of each transaction created by this database. See the transaction option description for more information." />
+            description="Set the maximum amount of backoff delay incurred in the call to ``onError`` if the error is retryable. This sets the ``max_retry_delay`` option of each transaction created by this database. See the transaction option description for more information."
+            defaultFor="502"/>
     <Option name="transaction_size_limit" code="503"
             paramType="Int" paramDescription="value in bytes"
-            description="Set the maximum transaction size which, if exceeded, will cause the transaction to be cancelled. Default to 10,000,000 bytes." />
+            description="Set the maximum transaction size in bytes. This sets the ``size_limit`` option on each transaction created by this database. See the transaction option description for more information." 
+            defaultFor="503"/>
     <Option name="snapshot_ryw_enable" code="26"
             description="Snapshot read operations will see the results of writes done in the same transaction. This is the default behavior." />
     <Option name="snapshot_ryw_disable" code="27"
```

Comment:
Snapshot RYW enable/disable I believe are also kind of like the other options in that they are set the database default (kind of). Do they also need `defaultFor` or something like it?

Reply:
I debated this one a bit. The reason I decided against it is that I opted to build in the assumption that with database defaults, only the most recently set value of the option matters. With these two options, that assumption doesn't hold.

Mainly the reason I went that route was to avoid having to store an entire history of database defaults. I was concerned that someone may write something that frequently changes the default for one option, and then I'd be stuck applying all values ever set for that option for every transaction creation and retry.

It may be desirable that I document this assumption somewhere. If you have any suggestions on where, I'd be happy to entertain them.

Reply:
Hm, I see. I think that's a reasonable enough assumption. I suppose that `getTransactionDefaults` or something could say this only includes options for which the last value is taken. It might also make sense to include a comment either here or in `DatabaseContext::setOption` that snapshot RYW enabling/disabling is handled differently because of the way it increments/decrements a counter rather than setting a value.